lib/helpers.bash

```diff
@@ -92,762 +96,742 @@ alias reload_completion="$(_make_reload_alias completion completion)"
+
+	local subdirectory="$1"
+	local file_type="$2"
+	local file_entity="$3"
+
+	if [[ -z "$file_entity" ]]; then
+		reference "disable-$file_type"
+		return
+	fi
+
+	local f suffix _bash_it_config_file plugins plugin
+	suffix="${subdirectory/plugins/plugin}"
+
+	if [[ "$file_entity" == "all" ]]; then
+		# Disable everything that's using the old structure
+
+		for _bash_it_config_file in "${BASH_IT}/$subdirectory/enabled"/*."${suffix}.bash"; do
+			rm -f "$_bash_it_config_file"
+		done
+
+		for _bash_it_config_file in "${BASH_IT}/enabled"/*".${suffix}.bash"; do
+			# Disable everything in the global "enabled" directory
+			rm -f "$_bash_it_config_file"
+		done
+	else
+		# Use a glob to search for both possible patterns
+		# 250---node.plugin.bash
+		# node.plugin.bash
+		# Either one will be matched by this glob
+		plugins=("${BASH_IT}/enabled"/*"${BASH_IT_LOAD_PRIORITY_SEPARATOR}${file_entity}.${suffix}.bash" "${BASH_IT}/$subdirectory/enabled/"{*"${BASH_IT_LOAD_PRIORITY_SEPARATOR}${file_entity}.${suffix}.bash","${file_entity}.${suffix}.bash"})
+		for plugin in "${plugins[@]}"; do
+			if [[ -e "${plugin}" ]]; then
+				rm "${plugin}"
+				plugin=
+				break
+			fi
+		done
+		if [[ -n "${plugin}" ]]; then
+			printf '%s\n' "sorry, $file_entity does not appear to be an enabled $file_type."
+			return
+		fi
+	fi
+
+	_bash-it-clean-component-cache "${file_type}"
+
+	printf '%s\n' "$file_entity disabled."
+}
+
+function _enable-plugin() {
+	_about 'enables bash_it plugin'
+	_param '1: plugin name'
+	_example '$ enable-plugin rvm'
+	_group 'lib'
+
+	_enable-thing "plugins" "plugin" "$1" "$BASH_IT_LOAD_PRIORITY_DEFAULT_PLUGIN"
+}
+
+function _enable-alias() {
+	_about 'enables bash_it alias'
+	_param '1: alias name'
+	_example '$ enable-alias git'
+	_group 'lib'
+
+	_enable-thing "aliases" "alias" "$1" "$BASH_IT_LOAD_PRIORITY_DEFAULT_ALIAS"
+}
+
+function _enable-completion() {
+	_about 'enables bash_it completion'
+	_param '1: completion name'
+	_example '$ enable-completion git'
+	_group 'lib'
+
+	_enable-thing "completion" "completion" "$1" "$BASH_IT_LOAD_PRIORITY_DEFAULT_COMPLETION"
+}
+
+function _enable-thing() {
+	cite _about _param _example
+	_about 'enables a bash_it component'
+	_param '1: subdirectory'
+	_param '2: file_type'
+	_param '3: file_entity'
+	_param '4: load priority'
+	_example '$ _enable-thing "plugins" "plugin" "ssh" "150"'
+
+	local subdirectory="$1"
+	local file_type="$2"
+	local file_entity="$3"
+	local load_priority="$4"
+
+	local _bash_it_config_file to_enable enabled_plugin enabled_plugin_global local_file_priority use_load_priority
+
+	if [[ -z "$file_entity" ]]; then
+		reference "enable-$file_type"
+		return
+	fi
+
+	if [[ "$file_entity" == "all" ]]; then
+		for _bash_it_config_file in "${BASH_IT}/$subdirectory/available"/*.bash; do
+			to_enable="$(basename "$_bash_it_config_file" ".$file_type.bash")"
+			if [[ "$file_type" == "alias" ]]; then
+				to_enable="$(basename "$_bash_it_config_file" ".aliases.bash")"
+			fi
```

Comment:
This can be simplified and get rid of the external program too. Though it's nice `basename` can truncate @(pre|post)fix at once, bash features are always preferable :grin: 

```suggestion
			: "${_bash_it_config_file##*/}"; to_enable="${_%.*.bash}"
```

Reply:
The particular code you suggest here won't work, but I agree 100% with your premise. I've alsö taken the opportunity to re-work the whole weirds with the not-globbing. Take another peek! 😃

Reply:
But I broke it, so I'll have to circle back this afternoon. oy!

Reply:
It's neat 👌